File: array/haskell/arr.hs
theinsert :: [a] -> Int -> a -> [a]
theinsert x i v
  | i >= length x = x
  | i < 0 = x
  | otherwise = theinsert' x i v
    where 
        theinsert' :: [a] -> Int -> a -> [a]
        theinsert' x 0 v = [v] ++ (tail x)
        theinsert' (x:xs) i v = [x] ++ theinsert' xs (i - 1) v

thedelete :: [a] -> Int -> [a]
thedelete x i
  | i >= length x = x
  | i < 0 = x
  | otherwise = thedelete' x i
    where
        thedelete' :: [a] -> Int -> [a]
        thedelete' x 0 = tail x
        thedelete' (x:xs) i = [x] ++ thedelete' xs (i - 1)

main = let arr = [1, 2, 3, 4, 5]
           idx = 2
           val = 20
        in putStrLn (show (theinsert arr idx val) ++ show (thedelete arr idx))

File: array/rust/arr.rs
const SIZE: i32 = 5
fn main() {
	let arr: [i32; SIZE] = [1, 2, 3, 4, 5]
}

fn insert(mut arr: [T; SIZE], idx: i32, val: T) -> [T; SIZE] {
    if idx >= SIZE || idx < 0 {
    }
}

fn main() {
    let m = 18;
    let n = 30;
    println!("{} と {} の最大公約数は {}", m, n, gcd(m, n));
}

fn gcd(m: i32, n: i32) -> i32 {
    if n == 0 {
        return m;
    }
    return gcd(n, m % n);
}
